manta-trusted-setup/src/groth16/ceremony/config/ppot.rs

```diff
@@ -371,6 +423,217 @@ where
+#[derive(Clone, Default)]
+pub struct Config(Ed25519<RawMessage<u64>>);
+
+impl Pairing for Config {
+    type Scalar = bn254::Fr;
+    type G1 = bn254::G1Affine;
+    type G1Prepared = <bn254::Bn254 as PairingEngine>::G1Prepared;
+    type G2 = bn254::G2Affine;
+    type G2Prepared = <bn254::Bn254 as PairingEngine>::G2Prepared;
+    type Pairing = bn254::Bn254;
+
+    #[inline]
+    fn g1_prime_subgroup_generator() -> Self::G1 {
+        bn254::G1Affine::prime_subgroup_generator()
+    }
+
+    #[inline]
+    fn g2_prime_subgroup_generator() -> Self::G2 {
+        bn254::G2Affine::prime_subgroup_generator()
+    }
+}
+
+impl Size for Config {
+    const G1_POWERS: usize = (Self::G2_POWERS << 1) - 1;
+    const G2_POWERS: usize = 1 << 17;
+}
+
+impl kzg::Configuration for Config {
+    type DomainTag = u8;
+    type Challenge = [u8; 64];
+    type Response = [u8; 64];
+    type HashToGroup = KZGBlakeHasher<Self>;
+
+    const TAU_DOMAIN_TAG: Self::DomainTag = 0;
+    const ALPHA_DOMAIN_TAG: Self::DomainTag = 1;
+    const BETA_DOMAIN_TAG: Self::DomainTag = 2;
+
+    #[inline]
+    fn hasher(domain_tag: Self::DomainTag) -> Self::HashToGroup {
+        Self::HashToGroup { domain_tag }
+    }
+
+    #[inline]
+    fn response(
+        state: &Accumulator<Self>,
+        challenge: &Self::Challenge,
+        proof: &kzg::Proof<Self>,
+    ) -> Self::Response {
+        let mut hasher = BlakeHasher::default();
+        for item in &state.tau_powers_g1 {
+            item.serialize_uncompressed(&mut hasher).unwrap();
+        }
+        for item in &state.tau_powers_g2 {
+            item.serialize_uncompressed(&mut hasher).unwrap();
+        }
+        for item in &state.alpha_tau_powers_g1 {
+            item.serialize_uncompressed(&mut hasher).unwrap();
+        }
+        for item in &state.beta_tau_powers_g1 {
+            item.serialize_uncompressed(&mut hasher).unwrap();
+        }
+        state.beta_g2.serialize_uncompressed(&mut hasher).unwrap();
+        hasher.0.update(challenge);
+        proof
+            .tau
+            .serialize(&mut hasher)
+            .expect("Consuming ratio proof of tau failed.");
+        proof
+            .alpha
+            .serialize(&mut hasher)
+            .expect("Consuming ratio proof of alpha failed.");
+        proof
+            .beta
+            .serialize(&mut hasher)
+            .expect("Consuming ratio proof of beta failed.");
+        into_array_unchecked(hasher.0.finalize())
+    }
+}
+
+impl StateType for Config {
+    type State = State<Self>;
+}
+
+impl ProofType for Config {
+    type Proof = Proof<Self>;
+}
+
+/// Challenge Type
+pub type Challenge = manta_util::Array<u8, 64>;
+
+impl<P> HashToGroup<P, Challenge> for BlakeHasher
+where
+    P: Pairing,
+    P::G2: Sample,
+{
+    #[inline]
+    fn hash(&self, challenge: &Challenge, ratio: (&P::G1, &P::G1)) -> P::G2 {
+        <Self as HashToGroup<P, [u8; 64]>>::hash(self, &challenge.0, ratio)
+    }
+}
```

Comment:
Can you move this to the same place that `BlakeHasher` is defined and generalize it to arbitrary size arrays?